### ISSUE TITLE
Allow user to join a new team via team invite link

### ISSUE THREAD

**Issue by jasonblais** [mattermost/desktop#420]
I confirm (by marking "x" in the [ ] below): 

- [X] This is not a troubleshooting question. [Troubleshooting questions go here: http://www.mattermost.org/troubleshoot/](http://www.mattermost.org/troubleshoot/).
- [X] This doesn't reproduce on web browsers (such as in Chrome). If it does, [issue reports go to the Mattermost Server repository](https://github.com/mattermost/platform/issues).
- [X] I have read [contributing guidelines](https://github.com/mattermost/desktop/blob/master/CONTRIBUTING.md). 

---

**Summary**
Allow user to join a new team via team invite link

**Steps to reproduce**

Windows 10 and Mac OSX 10.9 / Mattermost Server v3.6.1 / Mattermost Desktop v3.5.0

1) Join at least two teams on a server
2) Go to TEAM1, then Main Menu > Get Team Invite Link
3) Go to Main Menu > Leave Team
4) On the confirmation dialog, click "Yes"
5) Paste the link you copied in step 1 to a channel in TEAM 2
6) Click the link

**Expected behavior**
User joins the team and the team page opens for the user on the same window and tab.

**Observed behavior**
Win10: No action. The user does join the team but doesn't see it on his sidebar until refresh.
OSX 10.9: Team opens in a new app window.

**Comment by jasonblais** [comment]
I wasn't sure if this is a bug as it might have been caused by the team sidebar.

So added an improvement label for now.

**Comment by jnugh** [comment]
I created an issue in platform after trying to fix this issue: https://github.com/mattermost/platform/issues/6043

**Comment by jasonblais** [comment]
@jnugh Thanks for opening the issue in https://github.com/mattermost/platform/issues/6043

I think the issue description here is specific to the desktop app though, where clicking the team invite link doesn't work properly.

**Comment by jasonblais** [comment]
Chatted with @jnugh, this issue is indeed blocked on https://github.com/mattermost/platform/issues/6043

I've added server-side label for now and removed from the milestone until the issue is resolved.

**Comment by jasonblais** [comment]
A server-side PR was merged, but it looks like it didn't completely solve the issue. https://github.com/mattermost/platform/pull/6266

A follow-up PR is scheduled for later this month.

**Comment by jnugh** [comment]
Awesome, once the final fix is in I will prepare a fix for the desktop app :smile:

**Comment by jasonblais** [comment]
@jnugh The PR is here and now merged to /platform master branch: https://github.com/mattermost/platform/pull/6538

**Comment by jnugh** [comment]
Cool I'll start working on a PR this week :)

**Comment by jasonblais** [comment]
Actually, @jnugh, I just tested it and mattermost/platform#6538 fixes it for the desktop app too, at least on Windows 10.

1. Join at least two teams on a server
2. Go to TEAM1, then Main Menu > Get Team Invite Link
3. Go to Main Menu > Leave Team
4. On the confirmation dialog, click "Yes"
5. Paste the link you copied in step 1 to a channel in TEAM 2
6. Click the link

Expected behavior: User joins the team and the team page opens for the user on the same window and tab.

I've asked a team member to help try it on a Mac - if it's fixed there too, I'll close the issue.

**Comment by jasonblais** [comment]
Confirmed it's fixed on Mac as well, closing as resolved for Mattermost server 3.10 and later.
